parser/src/lib.rs

```diff
@@ -481,6 +481,22 @@ namespace N(2);
         assert_eq!(expected.trim(), printed.trim());
     }
 
+    #[test]
+    fn parse_impl() {
+        let input = r#"
+    impl<T> Iterator<ArrayIterator<T>, T> {
+        next: |it| if it.pos >= 7 { (it, none) } else { some((increment(it), it.arr[it.pos])) },
+    }"#;
+
+        let expected = r#"
+    impl<T> Iterator<ArrayIterator<T>, T> {
+        next: (|it| if it.pos >= 7 { (it, none) } else { some((increment(it), it.arr[it.pos])) }),
+    }"#;
+
```

Comment:
Can you add another one with `impl<A, B>`? Also do we already support bounds here like `impl<A: Eq, B: Add>`?

Reply:
Sure. No, we don't support bounds at the moment.